CMakeLists.txt

```diff
@@ -26,11 +27,16 @@ set(MLX_BUILD_TESTS OFF)
 set(MLX_BUILD_EXAMPLES OFF)
 set(MLX_BUILD_BENCHMARKS "Build benchmarks for mlx" OFF)
 set(MLX_BUILD_PYTHON_BINDINGS OFF)
-FetchContent_Declare(
-  mlx
-  GIT_REPOSITORY "https://github.com/ml-explore/mlx.git"
-  GIT_TAG v0.18.0)
-FetchContent_MakeAvailable(mlx)
+
+if(MLX_C_USE_SYSTEM_MLX)
+  find_package(MLX REQUIRED)
+else()
+  FetchContent_Declare(
+    mlx
+    GIT_REPOSITORY "https://github.com/ml-explore/mlx.git"
+    GIT_TAG v0.18.0)
```

Comment:
Thanks for the contribution. Could you rebase on main? (now we are fetching MLX v0.20.0)

Reply:
Done :-)